File: ABC/abc013/abc013_3 4.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    // 100点解法
    long long N, H, A, B, C, D, E;
    cin >> N >> H >> A >> B >> C >> D >> E;

    long long ans = N * C;
    for (long long i = 0; i <= N; i++) { // 普通の食事をとる日数
        for (long long j = 0; j <= N-i; j++) { // 質素な食事をとる日数
            if (H + B * i + D * j - E * (N - i - j) > 0) { // 満腹度が0以下にならないようにできる
                ans = min(ans, A * i + C * j);
            }
        }
    }

    cout << ans << endl;
}

File: ABC/abc013/abc013_3 4.py
#100点解法
N, H = map(int, input().split())
A, B, C, D, E = map(int, input().split())

ans = float("inf")
for i in range(N+1): #普通の食事をとる日数
    for j in range(N-i+1): #質素な食事をとる日数
        ans_ij = A * i + C * j #食費の合計
        s = H + B * i + D * j - E * (N - i - j) #満腹度
        if s > 0:
            ans = min(ans, ans_ij)

print(ans)
